pkg/apis/crd/v1alpha1/types.go

```diff
@@ -709,3 +709,152 @@ type ExternalNodeList struct {
+}
+
+type SupportBundleSpec struct {
+	Nodes         *BundleNodes          `json:"nodes,omitempty"`
+	ExternalNodes []BundleExternalNodes `json:"externalNodes,omitempty"`
+	// ExpirationMinutes is the requested duration of validity of the SupportBundleCollection.
+	// A SupportBundleCollection will be marked as Failed if it does not finish before expiration.
+	// Default is 60.
+	ExpirationMinutes int32 `json:"expirationMinutes"`
+	// SinceTime defines a duration newer than which the Agent logs are expected.
+	// A valid value is like: 1d, 2h, 30min.
+	SinceTime      string                        `json:"sinceTime,omitempty"`
+	FileServer     BundleFileServer              `json:"fileServer"`
+	Authentication BundleServerAuthConfiguration `json:"authentication"`
+}
+
+// BundlePhase defines the phase in which a SupportBundleCollection is.
+type BundlePhase string
+
+const (
+	// BundlePending means the SupportBundleCollection has been accepted by the system,
+	// but it has not been processed by Antrea.
+	BundlePending BundlePhase = "Pending"
+	// BundleProcessing means the SupportBundleCollection is being processed by Antrea.
+	BundleProcessing BundlePhase = "Processing"
+	// BundleProcessed means the SupportBundleCollection has been processed by all Nodes
+	// or ExternalNodes it requires.
+	BundleProcessed BundlePhase = "Processed"
+	// BundleFailed means the SupportBundleCollection failed. It can happen when any of
+	// the required Node or ExternalNode fails
+	// to process the request or gets timeout.
+	BundleFailed BundlePhase = "Failed"
+)
+
+// BundleFailedReason defines the reason why a SupportBundleCollection is marked as Failed.
+type BundleFailedReason string
+
+const (
+	// PartialSuccess means only some required Nodes or ExternalNodes succeed to process
+	// the SupportBundleCollection request before expiration, and others are failed.
+	PartialSuccess BundleFailedReason = "Partial success"
+	// ControllerError means an error occurs on Antrea Controller which breaks processing
+	// the SupportBundleCollection.
+	ControllerError BundleFailedReason = "Controller error"
+	// UnknownError means an unknown error occurs in Antrea when processing the
+	// SupportBundleCollection
+	UnknownError BundleFailedReason = "Unknown"
+)
+
+type SupportBundleStatus struct {
+	Phase BundlePhase `json:"phase"`
+	// The generation observed by Antrea.
+	ObservedGeneration int64 `json:"observedGeneration"`
+	// The time from which the support bundle request is started to be processed by Antrea.
+	StartTime metav1.Time `json:"startTime"`
+	// The number of Nodes and ExternalNodes that have processed the SupportBundleCollection.
+	CurrentNodes int32 `json:"currentNodes"`
+	// The total number of Nodes and ExternalNodes that should process the SupportBundleCollection.
+	DesiredNodes int32 `json:"desiredNodes"`
+	// The reason why the SupportBundleCollection phase is marked as Failed.
+	FailedReason BundleFailedReason `json:"failedReason,omitempty"`
```

Comment:
Should we use Conditions which can include more information?

Reply:
Added Condition